server/src/core/file_mgr.rs

```diff
@@ -397,24 +397,50 @@ impl FileInfo {
         Position::new(line as u32, column as u32)
     }
 
+    pub fn try_offset_to_position_with_rope(rope: &Rope, offset: usize) -> Option<Position> {
+        let char = rope.try_byte_to_char(offset).ok()?;
+        let line = rope.try_char_to_line(char).ok()?;
+        let first_char_of_line = rope.try_line_to_char(line).ok()?;
+        let column = char - first_char_of_line;
+        Some(Position::new(line as u32, column as u32))
+    }
+
     pub fn offset_to_position(&self, offset: usize) -> Position {
         FileInfo::offset_to_position_with_rope(self.file_info_ast.borrow().text_rope.as_ref().expect("no rope provided"), offset)
     }
 
+    pub fn try_offset_to_position(&self, offset: usize) -> Option<Position> {
+        FileInfo::try_offset_to_position_with_rope(self.file_info_ast.borrow().text_rope.as_ref().expect("no rope provided"), offset)
```

Comment:
Maybe you want to also not expect here? 

Perhaps:
```suggestion
        FileInfo::try_offset_to_position_with_rope(self.file_info_ast.borrow().text_rope.as_ref()?, offset)
```